src/setfit/data.py

```diff
@@ -15,69 +15,115 @@
+        in `dataset` containing the labels of the examples.
+        multi_label (`bool`, *optional*, defaults to `False`): Whether or not multiple
+        candidate labels can be true.
+        label_names_column (`str`, *optional*, defaults to "label_text"): The name of the
+        label column in the `reference_dataset`, to be used in case there is no ClassLabel
+        feature for the label column.
+
+    Returns:
+        `Dataset`: A copy of the input Dataset with templated examples added.
+
+    Raises:
+        `ValueError`: If the input Dataset is not empty and one or both of the
+            provided column names are missing.
+    """
+    if dataset is None:
+        dataset = Dataset.from_dict({})
+
+    required_columns = {text_column, label_column}
+    column_names = set(dataset.column_names)
+    if column_names:
+        missing_columns = required_columns.difference(column_names)
+        if missing_columns:
+            raise ValueError(f"The following columns are missing from the input dataset: {missing_columns}.")
+
+    if bool(reference_dataset) == bool(candidate_labels):
+        raise ValueError(
+            "Must supply exactly one of `reference_dataset` or `candidate_labels` to `get_templated_dataset()`!"
+        )
+
+    if candidate_labels is None:
+        candidate_labels = get_candidate_labels(reference_dataset, label_names_column)
+
+    empty_label_vector = [0] * len(candidate_labels)
+
+    for label_id, label_name in enumerate(candidate_labels):
+        label_vector = empty_label_vector.copy()
+        label_vector[label_id] = 1
+        example = {
+            text_column: template.format(label_name),
+            label_column: label_vector if multi_label else label_id,
         }
-    else:
-        raise ValueError(f"Dataset {dataset} not supported for data augmentation!")
+        for _ in range(sample_size):
+            dataset = dataset.add_item(example)
+
+    return dataset
+
+
+def get_candidate_labels(dataset_name: str, label_names_column: str = "label_text") -> List[str]:
+    dataset = load_dataset(dataset_name, split="train")
+
+    try:
+        # Extract ClassLabel feature from "label" column
+        label_features = dataset.features["label"]
+        # Label names to classify with
+        candidate_labels = label_features.names
+
+    except AttributeError:
+        # Some datasets on the Hugging Face Hub don't have a ClassLabel feature for the label column.
+        # In these cases, you should compute the candidate labels manually by first computing the id2label mapping.
+
+        # The column with the label names
+        label_names = dataset.unique(label_names_column)
+        # The column with the label IDs
+        label_ids = dataset.unique("label")
+        id2label = dict(zip(label_ids, label_names))
+        # Sort by label ID
+        id2label_sorted = {key: val for key, val in sorted(id2label.items(), key=lambda x: x[0])}
```

Comment:
Can this not be simplified to something like
```python
id2label_sorted = dict(sorted(id2label.items(), key=lambda x: x[0]))
```
I can't test this right now, but I think it's possible. 

Also, you can probably just
```python
_, candidate_labels = zip(*sorted(id2label.items(), key=lambda x: x[0]))
```
But that's also off the top of my head, not 100% sure if that'd work.

Reply:
Thanks @tomaarsen for all your important comments  :)
I simplified it without using `dict`.
Ready for review.